src/client/tls.rs

```diff
@@ -132,7 +119,7 @@ impl<S> TlsPreloginWrapper<S> {
 }
 
 #[cfg(feature = "tls")]
-impl<S: AsyncRead + AsyncWrite + Unpin> AsyncRead for TlsPreloginWrapper<S> {
+impl<S: futures::AsyncRead + futures::AsyncWrite + Unpin + Send> futures::AsyncRead for TlsPreloginWrapper<S> {
```

Comment:
You might want to use futures::{AsyncRead, AsyncWrite} to save valuable horizontal space.